lib/msal-browser/src/interaction_client/SilentAuthCodeClient.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AuthenticationResult, ICrypto, Logger, PromptValue, CommonAuthorizationCodeRequest, AuthorizationCodeClient, AuthError } from "@azure/msal-common";
+import { StandardInteractionClient } from "./StandardInteractionClient";
+import { AuthorizationUrlRequest } from "../request/AuthorizationUrlRequest";
+import { BrowserConfiguration } from "../config/Configuration";
+import { BrowserCacheManager } from "../cache/BrowserCacheManager";
+import { EventHandler } from "../event/EventHandler";
+import { INavigationClient } from "../navigation/INavigationClient";
+import { BrowserAuthError } from "../error/BrowserAuthError";
+import { InteractionType, ApiId } from "../utils/BrowserConstants";
+import { SilentHandler } from "../interaction_handler/SilentHandler";
+import { SsoSilentRequest } from "../request/SsoSilentRequest";
+
+export class SilentAuthCodeClient extends StandardInteractionClient {
+    private apiId: ApiId;
+
+    constructor(config: BrowserConfiguration, storageImpl: BrowserCacheManager, browserCrypto: ICrypto, logger: Logger, eventHandler: EventHandler, navigationClient: INavigationClient, apiId: ApiId, correlationId?: string) {
+        super(config, storageImpl, browserCrypto, logger, eventHandler, navigationClient, correlationId);
+        this.apiId = apiId;
+    }
+    
+    /**
+     * Acquires a token silently by opening a hidden iframe to the /authorize endpoint with prompt=none
+     * @param request 
+     */
+    async acquireToken(request: SsoSilentRequest): Promise<AuthenticationResult> {
+        this.logger.trace("SilentAuthCodeClient.acquireToken called");
+
+        // Check that prompt is set to none, throw error if it is set to anything else.
+        if (request.prompt && request.prompt !== PromptValue.NONE) {
+            throw BrowserAuthError.createSilentPromptValueError(request.prompt);
+        }
+
+        // Auth code payload is required
+        if (!request.authCodePayload || !request.authCodePayload.code) {
+            throw "";
+        }
+
```

Comment:
TODO: add error checking for nonce

Reply:
Nevermind, we will not be doing nonce validation for this flow (as it is not needed, given no redirect is happening).